File: common-core/src/main/java/io/quartic/common/SweetStyle.java
package io.quartic.common;

import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import org.immutables.value.Value;
import org.immutables.value.Value.Style.ImplementationVisibility;

@JsonSerialize
@Value.Style(
        add = "*",
        addAll = "all*",
        depluralize = true,
        allParameters = true,
        jdkOnly = true,
        typeAbstract = "Abstract*",
        typeImmutable = "*",
        visibility = ImplementationVisibility.PUBLIC)
public @interface SweetStyle {
}

File: common-core/src/main/kotlin/io/quartic/common/SweetStyle.kt
package io.quartic.common

import org.immutables.value.Value
import org.immutables.value.Value.Style.ImplementationVisibility

@Value.Style(
        add = "*",
        addAll = "all*",
        put = "*",
        putAll = "all*",
        depluralize = true,
        allParameters = true,
        jdkOnly = true,
        typeImmutable = "*Impl",
        visibility = ImplementationVisibility.PUBLIC
)
annotation class SweetStyle
